File: sol/solution/0001-0100/0014. Longest Common Prefix/Solution.scala
object Solution {
    def longestCommonPrefix(strs: Array[String]): String = {
      if (strs.isEmpty) return ""
      var prefix = strs(0)
      for(i <- 1 until strs.length)
      {
        while(!strs(i).startsWith(prefix) && prefix.nonEmpty)
        {
          prefix = prefix.substring(0,prefix.length-1)
        }
      }
      prefix
    }
}

File: sol/solution/0001-0100/0014. Longest Common Prefix/Solution.dart
class Solution {
  String longestCommonPrefix(List<String> strs) {
    if (strs.isEmpty) return "";
    String prefix = strs[0];
    for (var str in strs.skip(1)) {
      while (!str.startsWith(prefix)) {
        prefix = prefix.substring(0, prefix.length - 1);
        if (prefix.isEmpty) return "";
      }
    }
    return prefix;
  }
}